shell/browser/api/electron_api_app.cc

```diff
@@ -1442,6 +1442,26 @@ void App::SetUserAgentFallback(const std::string& user_agent) {
   ElectronBrowserClient::Get()->SetUserAgent(user_agent);
 }
 
+#if defined(OS_WIN)
+
+bool App::IsRunningUnderARM64Translation() const {
+  USHORT processMachine = 0;
+  USHORT nativeMachine = 0;
+
+  auto IsWow64Process2 = reinterpret_cast<decltype(&::IsWow64Process2)>(
+      GetProcAddress(GetModuleHandle(L"kernel32.dll"), "IsWow64Process2"));
+
+  if (IsWow64Process2) {
+    if (!IsWow64Process2(GetCurrentProcess(), &processMachine,
+                         &nativeMachine)) {
+      return false;
+    }
+  }
+
+  return nativeMachine == IMAGE_FILE_MACHINE_ARM64;
```

Comment:
Shouldn’t you also check if processMachine is not arm64?

Reply:
AFAIK that shouldn't be necessary because if both process and OS are arm64, WOW is not used (there is nothing to translate), and therefore `IsWow64Process2` doesn't succeed.

I tested this code with arm64 builds on arm64 machines, x64 builds on arm64 machines and x64 builds on x64 builds on x64 machines, and I got the expected result (only x64 builds on arm64 machines return `true` for this function).